app/web/features/profile/view/UserTextAndLabel.tsx

```diff
@@ -41,6 +46,60 @@ export const LabelsReferencesLastActive = ({ user }: Props) => (
   </>
 );
 
+export const LabelsResponseRate = ({ user }: Props) => {
+  const { t } = useTranslation("profile");
+  const query = useQuery(responseRateKey(user.userId), () =>
+    service.requests.getResponseRate(user.userId)
+  );
+  const rateText =
+    query.data?.responseRate === ResponseRate.RESPONSE_RATE_INSUFFICIENT_DATA
+      ? t("response_rate_text_insufficient")
+      : query.data?.responseRate === ResponseRate.RESPONSE_RATE_LOW
+      ? t("response_rate_text_low")
+      : query.data?.responseRate === ResponseRate.RESPONSE_RATE_SOME
+      ? t("response_rate_text_some")
+      : query.data?.responseRate === ResponseRate.RESPONSE_RATE_MOST
+      ? t("response_rate_text_most")
+      : query.data?.responseRate === ResponseRate.RESPONSE_RATE_ALMOST_ALL
+      ? t("response_rate_text_almost_all")
+      : undefined;
+  const timeText =
+    query.data?.responseRate === ResponseRate.RESPONSE_RATE_SOME
+      ? t("response_time_text_some", {
+          p33: dayjs
+            .duration(query.data.responseTimeP33!.seconds, "second")
+            .humanize(),
+        })
+      : query.data?.responseRate === ResponseRate.RESPONSE_RATE_MOST
+      ? t("response_time_text_most", {
+          p33: dayjs
+            .duration(query.data.responseTimeP33!.seconds, "second")
+            .humanize(),
+          p66: dayjs
+            .duration(query.data.responseTimeP66!.seconds, "second")
+            .humanize(),
+        })
+      : query.data?.responseRate === ResponseRate.RESPONSE_RATE_ALMOST_ALL
+      ? t("response_time_text_almost_all", {
+          p33: dayjs
+            .duration(query.data.responseTimeP33!.seconds, "second")
+            .humanize(),
+          p66: dayjs
+            .duration(query.data.responseTimeP66!.seconds, "second")
+            .humanize(),
+        })
+      : undefined;
+
+  return (
+    <>
+      <LabelAndText label={t("response_rate_label")} text={rateText ?? ""} />
+      {timeText && (
+        <LabelAndText label={t("response_time_label")} text={timeText} />
```

Comment:
What about if `timeText` is undefined? Maybe in the ternary chains above, the last `else`s can fallback to the empty string instead so we don't need to use `??` here (same comment with `rateText`)?

Reply:
`timeText` is undefined when the rate text is "not enough data" or "rarely responds" or is loading.
`rateText` is only undefined when the user is loading. Maybe `"..."` is a better string? Is there a better way to make this better/more clearly typed?

Reply:
Yeah, can't really think of anything more succinct than what you got already since it's impossible to know the value without checking through all the possible enum values.

Maybe for `timeText` /response time, when the user "rarely responds" or "not enough data", we should either have a fallback to something like "unknown" (or a better message) or hide that label altogether? Otherwise, I think it's a bit confusing to have "Response time: " with no value shown next to it.

Reply:
Only thing I can think of is do the check once with the tweaked API response and set both text in the checks? I think if statements/switches are ok as long they are used after the hooks, but I guess not really ideal if you always have to remember the order of where you put the hooks.

Could also contain them in two separate components. That doesn't necessarily reduce the verbosity but at least the logic is grouped together with the UI code that needs it straight after